cirq/sim/sparse_simulator.py

```diff
@@ -331,3 +336,27 @@ def sample(self,
                                                     self, None),
                                                 repetitions=repetitions,
                                                 seed=seed)
+
+
+def _split_into_unitary_then_general(circuit: 'cirq.Circuit'
+                                    ) -> Tuple['cirq.Circuit', 'cirq.Circuit']:
```

Comment:
Ditto.  I would either expand on the comment above "Simulate as many unitary operations as possible..." or write a docstring to explain what this is doing and returning.

Reply:
Acknowledged. In this case I think the method name is pretty descriptive.

Reply:
Added a docstring.